src/a-runtime-console/kubernetes/ui/pipeline/input-action-dialog/input-action-dialog.component.ts

```diff
@@ -84,4 +88,76 @@ export class InputActionDialog {
   close() {
     this.modal.close();
   }
+
+  checkIfJenkinsIsUpAndProceedURL() {
+    let notification;
+    this.callJenkins()
+      .subscribe((response) => {
+        switch (response.status) {
+          case 200:
+            notification = {
+              message: `Got ${response.status}, Jenkins is up and running.`,
+              type: NotificationType.SUCCESS
+            };
+            this.notifications.message(notification as Notification);
+            this.invokeUrl(this.inputAction.proceedUrl);
+            return;
+          case 307:
+          case 302:
+            notification = {
+              message: `Got ${response.status}, Connecting to Jenkins.`,
+              type: NotificationType.INFO
+            };
+            break;
+          case 503:
+            notification = {
+              message: `Got ${response.status}, Cluster resources capacity is full. Waiting along.`,
+              type: NotificationType.INFO
+            };
+            break;
+          case 504:
+            notification = {
+              message: `Got ${response.status}, Request timed out. Trying again.`,
+              type: NotificationType.WARNING
+            };
+            break;
+          case 202:
+            notification = {
+              message: `Got ${response.status}, Jenkins is currently idled. Please wait while we start it. waiting along.`,
+              type: NotificationType.WARNING
+            };
+            break;
+          default:
+            notification = {
+              message: `Got status ${response.status}`,
+              type: NotificationType.DANGER
+            };
+            this.notifications.message(notification as Notification);
+            return;
+        }
+        this.notifications.message(notification as Notification);
+        setTimeout (this.checkIfJenkinsIsUpAndProceedURL(), 10000);
+      });
+    this.close();
+  }
+
+  callJenkins(): Observable<Response> {
+    const token_json = {
+      access_token: this.authService.getToken(),
+      token_type: 'Bearer'
+    };
+    let url = this.build.jenkinsBuildURL + '?token_json=' + encodeURIComponent(JSON.stringify(token_json));
+    return this.http
+      .get(url)
+      .map((response: Response) => {
```

Comment:
Using `map` to take `response` and return` response` is redundant, `http.get()` already returns the Response object, no?

`http.get(...).error(...)` should do the same thing.